src/core/python/properties_v.cpp

```diff
@@ -41,10 +47,14 @@ py::object properties_get(const Properties& p, const std::string &key) {
         return py::cast(p.get<Color<PFloat, 3>>(key));
     else if (type == Properties::Type::Array3f)
         return py::cast(p.get<dr::Array<PFloat, 3>>(key));
-    else if (type == Properties::Type::Transform)
+    else if (type == Properties::Type::Transform3f)
+        return py::cast(p.get<Transform<Point<PFloat, 3>>>(key));
+    else if (type == Properties::Type::Transform4f)
         return py::cast(p.get<Transform<Point<PFloat, 4>>>(key));
     // else if (type == Properties::Type::AnimatedTransform)
         // return py::cast(p.animated_transform(key));
+    else if (type == Properties::Type::Tensor)
+        return py::cast(*(TensorXf*)(p.get<TensorHandle>(key).get()));
```

Comment:
Could be worth printing a verbose error message here if the TensorHandle points to a tensor type of a different variant.